store-implementation/federated-store/src/test/java/uk/gov/gchq/gaffer/federatedstore/FederatedStoreSchemaTest.java

```diff
@@ -70,95 +94,210 @@ public void setUp() throws Exception {
+                .first(new GetAllElements.Builder()
+                        //No view so makes default view, should get only view compatible with graph "a"
+                        .option(FederatedStoreConstants.KEY_OPERATION_OPTIONS_GRAPH_IDS, "a")
+                        .build())
+                .build(), testContext);
 
-        // No exceptions thrown - as all 3 graphs should be able to be added together.
+        // Then
+        assertNotNull(a);
+        assertFalse(a.iterator().hasNext());
     }
 
     @Test
-    public void shouldGetCorrectDefaultViewForAChosenGraphOperation() throws
-            Exception {
-        final HashMapGraphLibrary library = new HashMapGraphLibrary();
-        library.add("a", new Schema.Builder()
-                .edge("e1", getProp("prop1"))
-                .type(DIRECTED_EITHER, Boolean.class)
-                .merge(STRING_SCHEMA)
-                .build(), PROPERTIES);
+    public void shouldBeAbleToGetElementsWithOverlappingSchemas() throws
+            OperationException {
+        // Given
+        addOverlappingPropertiesGraphs(false);
 
-        library.add("b", new Schema.Builder()
-                .edge("e1", getProp("prop2"))
-                .type(DIRECTED_EITHER, Boolean.class)
-                .merge(STRING_SCHEMA)
-                .build(), PROPERTIES);
+        // Element 1
+        fStore.execute(new AddElements.Builder()
+                .input(new Edge.Builder()
+                        .group("e1")
+                        .source("source1")
+                        .dest("dest1")
+                        .property("prop1", "value1")
+                        .build())
+                .build(), testContext);
 
+        // Element 2
+        fStore.execute(new AddElements.Builder()
+                .input(new Edge.Builder()
+                        .group("e1")
+                        .source("source1")
+                        .dest("dest2")
+                        .property("prop1", "value1")
+                        .property("prop2", "value2")
+                        .build())
+                .build(), testContext);
 
-        fStore.execute(new AddGraph.Builder()
-                .graphId("a")
+        // When
+        final CloseableIterable<? extends Element> a = fStore.execute(new GetElements.Builder()
+                .input(new EntitySeed("source1"))
+                .view(new View.Builder()
+                        .edge("e1", new ViewElementDefinition.Builder()
+                                .properties("prop2")
+                                .build())
+                        .build())
                 .build(), testContext);
 
-        fStore.execute(new AddGraph.Builder()
-                .graphId("b")
+        assertNotNull(a);
+        final Set<? extends Element> resultsSet = Streams.toStream(a).collect(Collectors.toSet());
+        final List<? extends Element> resultsList = Streams.toStream(a).collect(Collectors.toList());
```

Comment:
why lists and sets?
Use Lists it won't dedup

Reply:
Sets are used to compare expected and results regardless of order